### ISSUE TITLE
Comparing two identical arrays

### ISSUE THREAD

**Issue by 1am** [willhoag/array-compare#2]
Hi,

I've been testing out your library and ran into some strange behaviour when comparing 2 identical arrays. I'll demonstrate this with these code snippets:

### Compare 2 identical arrays 

```
var ac = require('array-compare');

var arrayA = [
  { sandwich: 'turkey', good: 'no' },
  { sandwich: 'blt', good: 'yes' },
];

var arrayB = [
  { sandwich: 'turkey', good: 'no' },
  { sandwich: 'blt', good: 'yes' },
];

var r = ac(arrayA, arrayB);
console.log('found:', r.found);
console.log('missing:', r.missing);
console.log('added:', r.added);
```

Output from this script is the following:

```
found: [ { a: { sandwich: 'turkey', good: 'no' },
    b: { sandwich: 'turkey', good: 'no' } },
  { a: { sandwich: 'blt', good: 'yes' }, b: undefined } ]
missing: []
added: [ { b: { sandwich: 'blt', good: 'yes' } } ]
```

### Adding an empty object at the end of the array makes it work fine:

```
var ac = require('array-compare');

var arrayA = [
  { sandwich: 'turkey', good: 'no' },
  { sandwich: 'blt', good: 'yes' },
  {}
];

var arrayB = [
  { sandwich: 'turkey', good: 'no' },
  { sandwich: 'blt', good: 'yes' },
  {}
];

var r = ac(arrayA, arrayB);
console.log('found:', r.found);
console.log('missing:', r.missing);
console.log('added:', r.added);
```

output:

```
found: [ { a: { sandwich: 'turkey', good: 'no' },
    b: { sandwich: 'turkey', good: 'no' } },
  { a: { sandwich: 'blt', good: 'yes' }, b: {} },
  { a: {}, b: { sandwich: 'blt', good: 'yes' } } ]
missing: []
added: []
```

Tested it with nodejs 6.9.1, 4.4.6 and 0.12.7 with the same results. I've looked into the code but couldn't really catch where the issue comes from. Is there any chance you would have an idea on how to fix this?

**Comment by willhoag** [comment]
Found this bug, in addition another bug and will fix them both. The implementation is intended to match either by object key (id) or by reference, so in your examples, because you didn't include a key to the 3rd parameter (id), it should have compared by reference and resulted in non being found.

Because of how egregious the bugs were, the results were quite different. I can't tell you how I fowled up so much here, but the following fix should make it work as intended. So, there is no deep-equal functionality.

A better implementation would let you provide your own comparing function. Might make this into a new major version or break it off into a separate module.
